### ISSUE TITLE
Unsigned upload doesn't timeout 

### ISSUE THREAD

**Issue by QuyHd123** [cloudinary/cloudinary_ios#396]
## Bug report for Cloudinary iOS SDK
Before proceeding, please update to latest version and test if the issue persists

## Describe the bug in a sentence or two.
When the `upload` function is called with the `timeout` parameter set to 30000 and there is no internet connection, the upload function `completionHandler` callback doesn't get called.

## Issue Type (Can be multiple)
[ ] Build - Can’t install or import the SDK
[ ] Performance - Performance issues
[x] Behaviour - Functions aren’t working as expected (Such as generate URL)
[ ] Documentation - Inconsistency between the docs and behaviour
[ ] Other (Specify)

## Steps to reproduce
Setup:
```
    private lazy var cloudinary: CLDCloudinary = {
        let config = CLDConfiguration(cloudName: cloudName, timeout: 30000)
        return CLDCloudinary(configuration: config)
    }()
```
Upload process:
```
        cloudinary
            .createUploader()
            .upload(data: imageData, uploadPreset: uploadPreset, completionHandler: { [weak self] result, error in
                if let result = result {
                    print("Upload success, url: \(String(describing: result.url))")
                }
                if let error = error {
                    print(error.localizedDescription)
                }
            })
```
## Error screenshots or Stack Trace (if applicable)
…

## Build/Dependency management
[x] Cocoa-Pods
[ ] Carthage
[ ] Manual import
[ ] Other (Specify)

## Is the issue reproducible only on a specific device?
[x] No
[ ] Yes (specify model + iOS version)

## Versions and Libraries (fill in the version numbers)
iOS Cloudinary SDK version - 3.3.0
OSX (on the dev environment) - 13.0 (22A380)
XCode - 14.2 (14C18)
Swift - 5.7.2
Target iOS - 13.0

Repository
If possible, please provide a link to a reproducible repository that showcases the problem

**Comment by QuyHd123** [comment]
@atdcloud I don't mind, this is the value I used:
  - cloudname: "dongphamcom"
  - uploadpreset: "sbjytzqx"

**Comment by atdcloud** [comment]
@QuyHd123 Thank you.  I retested the `completionHandler` only this time without an active network to the internet and I was able to reproduce the error.  The `completionHandler` did not get triggered even though the upload processed elapsed more than 30 seconds and `timeout` is set to 30 seconds in the initial configuration.  I have raised this issue to our internal development team and we will push the update.
